### ISSUE TITLE
Trace.resample not setting stats.sampling_rate correctly, if number of points is not an integer multiple of the decimation factor

### ISSUE THREAD

**Issue by kaeufl** [obspy/obspy#857]
Maybe I'm missing something here, but it seems to me as if in Trace.resample(sampling_rate) the Trace.stats.sampling_rate is not properly set in the case that 

``` python
len(tr.data) % (tr.stats.sampling_rate / float(sampling_rate)) != 0.0
```

From the docs of scipy.signal.resample:

> The first sample of the returned vector is the same as the first
> sample of the input vector.  The spacing between samples is changed
> from dx to:
> 
>    dx \* len(x) / num

That is, as far as I understand the docs, the signal x is treated as having fixed length and resampled at num new points. If the condition len(tr.data) % factor != 0.0 is not met, then the actual sampling rate might therefore be different from the desired sampling rate.

However, in Trace.resample (https://github.com/obspy/obspy/blob/0.9.2/obspy/core/trace.py#L1468),
Trace.stats.sampling_rate is set to with whatever argument the method has been called, as far as I can see.

Thanks for clarification, if I'm wrong. Otherwise, I'd be happy to write a pull request.

**Comment by s-g-mo** [comment]
I'm not sure if this has already been addressed, but something similar is occurring with the tr.decimate() function. When using this function in v.1.0.0 the new sampling rate isn't updated in the trace metadata as well. I've tried to work around it by decimating, then forcing the trace to update the stats with the new sampling rate using tr.stats.update. This does work for the current instance, however, when writing the trace out as a .SAC, the old sampling rate is written into the stats.
